example_config_class.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+class ThingWorker
+  def self.process(message)
+    FastlyNsq.logger.info("IN PROCESS: #{message}")
+  end
+end
+
+FastlyNsq.configure do |config|
+  config.channel = 'william'
+  # config.logger = Logger.new
+  # config.preprocessor = ->(message) { FastlyNsq.logger.info 'PREPROCESSESES' }
+
+  config.listen_to do |topics|
+    topics.add('assign_supplemental_plans', ThingWorker)
+  end
```

Comment:
This is pulled from Sidekiq.

The configure part seem to be build up this way so you can have several calls out to configure while init is happening.
```
FastlyNsq.configure do |config|
  config.channel = 'william'
  config.logger = SomeLogger
end

# Some time later
FastlyNsq.configure do |config|
  config.listen_to do |topics|
    topics.add('thing', ThingProcessor)
  end
end
```

The block mechanism for adding topic => Worker things is again seemingly built up to allow you to additively construct your hash. You could do this by passing hashes as well and just merging hashes... this is once again a style pulled from PerodicJobs in Sidekiq... You could say that it is slightly easier to validate, log, or do other actions on each added item rather than to loop over a merged hash each time...